pkg/provider/keepersecurity/client.go

```diff
@@ -0,0 +1,468 @@
+	errKeeperSecurityInvalidSecretDuplicatedKey = "Invalid Secret. Keys are not unique"
+	errKeeperSecurityInvalidProperty            = "Invalid Property. Secret %s does not have any key matching %s"
+	errKeeperSecurityInvalidField               = "Invalid Field. Key %s does not exists"
+	errKeeperSecurityNoFields                   = "Invalid Secret. Secret %s does not contain any valid field/file"
+	keeperSecurityFileRef                       = "fileRef"
+	keeperSecurityMfa                           = "oneTimeCode"
+	errTagsNotImplemented                       = "'find.tags' is not implemented in the KeeperSecurity provider"
+	errPathNotImplemented                       = "'find.path' is not implemented in the KeeperSecurity provider"
+	errInvalidJsonSecret                        = "Invalid Secret. Secret %s can not be converted to JSON. %w"
+	errInvalidRegex                             = "find.name.regex. Invalid Regular expresion %s. %w"
+	errInvalidRemoteRefKey                      = "match.remoteRef.remoteKey. Invalid format. Format should match secretName/key got %s"
+	errInvalidSecretType                        = "ESO can only push/delete %s record types. Secret %s is type %s"
+	errFieldNotFound                            = "Secret %s does not contain any custom field with label %s"
+
+	externalSecretType = "externalSecrets"
+	secretType         = "secret"
+	LoginType          = "login"
+	LoginTypeExpr      = "login|username"
+	PasswordType       = "password"
+	UrlTypeExpr        = "url|baseurl"
+	UrlType            = "url"
+)
+
+type Client struct {
+	ksmClient KeeperSecurityClient
+	folderID  string
+}
+
+type KeeperSecurityClient interface {
+	GetSecrets(filter []string) ([]*ksm.Record, error)
+	GetSecretByTitle(recordTitle string) (*ksm.Record, error)
+	CreateSecretWithRecordData(recUid, folderUid string, recordData *ksm.RecordCreate) (string, error)
+	DeleteSecrets(recrecordUids []string) (map[string]string, error)
+	Save(record *ksm.Record) error
+}
+
+type KeeperSecurityField struct {
+	Type  string   `json:"type"`
+	Value []string `json:"value"`
+}
+
+type KeeperSecurityCustomField struct {
+	Type  string   `json:"type"`
+	Label string   `json:"label"`
+	Value []string `json:"value"`
+}
+
+type KeeperSecurityFile struct {
+	Title   string `json:"type"`
+	Content string `json:"content"`
+}
+
+type KeeperSecuritySecret struct {
+	Title  string                      `json:"title"`
+	Type   string                      `json:"type"`
+	Fields []KeeperSecurityField       `json:"fields"`
+	Custom []KeeperSecurityCustomField `json:"custom"`
+	Files  []KeeperSecurityFile        `json:"files"`
+}
+
+func (c *Client) Validate() (esv1beta1.ValidationResult, error) {
+
+	return esv1beta1.ValidationResultReady, nil
+}
+
+func (c *Client) GetSecret(ctx context.Context, ref esv1beta1.ExternalSecretDataRemoteRef) ([]byte, error) {
+	record, err := c.findSecretById(ref.Key)
```

Comment:
i do think it is better to update the docs and point it to the record UID as it is the unique identifier of the record. Name is not unique, it can produce some errors. Name should be used only with  `dataFrom` and regex imho

Reply:
Sure, that works for me!